tiledb/sm/query/query_condition.cc

```diff
@@ -1470,13 +1516,87 @@ Status QueryCondition::apply_clause_sparse(
     case Datatype::STRING_UCS2:
     case Datatype::STRING_UCS4:
     default:
-      return Status_QueryConditionError(
-          "Cannot perform query comparison; Unsupported query "
-          "conditional type on " +
-          clause.field_name_);
+      throw std::runtime_error(
+          "Cannot perform query comparison; Unsupported query conditional type "
+          "on " +
+          node->get_field_name());
   }
+}
 
-  return Status::Ok();
+template <typename BitmapType, typename CombinationOp>
+void QueryCondition::apply_tree_sparse(
+    const tdb_unique_ptr<ASTNode>& node,
+    const ArraySchema& array_schema,
+    ResultTile& result_tile,
+    CombinationOp combination_op,
+    std::vector<BitmapType>& result_bitmap) const {
+  // For each child, declare a new result buffer, combine it all into the
+  // other one based on the combination op.
+  const QueryConditionCombinationOp& c_op = node->get_combination_op();
+  const uint8_t fill_value = c_op == QueryConditionCombinationOp::AND ? 1 : 0;
+  std::vector<BitmapType> combination_op_bitmap(
+      result_bitmap.size(), fill_value);
+
+  for (const auto& child : node->get_children()) {
```

Comment:
No, it should never reach here, because I test in apply_sparse whether something is originally a value node, and then I run it with this function which should cover all expression trees correctly. I can add a check and throw std::logic_error though!